omise/views/templates/hook/1.7/card_payment.tpl

```diff
@@ -65,74 +65,53 @@
-      }
 
-      return false;
-    };
-
-    var lockOmiseCardPaymentForm = function lockOmiseCardPaymentForm(form) {
-      form.name.disabled = true;
-      form.number.disabled = true;
-      form.expiration_month.disabled = true;
-      form.expiration_year.disabled = true;
-      form.security_code.disabled = true;
-    };
-
-    var omiseCardPaymentForm = {
-      name: document.getElementById('omise_card_holder_name'),
-      number: document.getElementById('omise_card_number'),
-      expiration_month: document.getElementById('omise_card_expiration_month'),
-      expiration_year: document.getElementById('omise_card_expiration_year'),
-      security_code: document.getElementById('omise_card_security_code'),
-    };
-
-    var omiseCreateTokenCallback = function omiseCreateTokenCallback(statusCode, response) {
-      if (statusCode === 200) {
-        document.getElementById('omise_card_token').value = response.id;
-        document.getElementById('omise_checkout_form').submit();
-      } else {
-        alert(response.message);
-        unlockOmiseCardPaymentForm(omiseCardPaymentForm);
-        paymentConfirmationButton.disabled = false;
-        paymentConfirmationButton.innerHTML = originalPaymentConfirmationText;
-      }
-    };
-
-    var unlockOmiseCardPaymentForm = function unlockOmiseCardPaymentForm(form) {
-      form.name.disabled = false;
-      form.number.disabled = false;
-      form.expiration_month.disabled = false;
-      form.expiration_year.disabled = false;
-      form.security_code.disabled = false;
-    };
-
-    document.addEventListener('DOMContentLoaded', function () {
-      paymentConfirmationButton = document.getElementById('payment-confirmation').getElementsByTagName('button')[0];
-      originalPaymentConfirmationText = paymentConfirmationButton.innerHTML;
-
-      paymentConfirmationButton.addEventListener('click', function (event) {
-        if (isOmiseCardPaymentOptionSelected()) {
+    var
+      originalPaymentConfirmationText,
+      paymentConfirmationButton,
+      $ = document.getElementById.bind(document),
+
+      omiseCardFields = [
+        'name',
+        'number',
+        'expiration_month',
+        'expiration_year',
+        'security_code'
+      ],
+
+      omiseCardPaymentForm = omiseCardFields.reduce(function(obj, field) {
+        obj[field] = $('omise_card_' + field);
+        return obj;
+      }, {}),
+
+      createOmiseToken = function createOmiseToken(form) {
+        var card = omiseCardFields.reduce(function(obj, field) { obj[field] = form[field].value; return obj;}, {});
+        Omise.setPublicKey('{$omise_public_key}');
+        Omise.createToken('card', card, omiseCreateTokenCallback);
+      },
+
+      isOmiseCardPaymentOptionSelected = function isOmiseCardPaymentOptionSelected() {
+        return document.querySelector('[data-module-name="omise-card-payment"]').checked;
+      },
+
+      setPayFormLockState = function setPayFormLockState(form, state) {
```

Comment:
Wonder if the name would be better with `setCardFormLockedState` the same as in template 1.6